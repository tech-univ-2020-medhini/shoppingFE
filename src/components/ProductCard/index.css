.product-card{
    display: flex;
    flex-direction: column;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    height: 20rem;
    width: 20rem;
    font-size: small;
    border: groove;
    background-color: rgb(255, 255, 255);
    border-radius: 10%;
    margin-top: 3rem;
    padding-left: 1rem;
    padding-right: 1rem;
}
.product-card:hover{
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}
img{
    align-self: center;
    max-height: 50%;
    max-width: 50%;
    margin: 1rem;
    border-radius: 10%;
    border: groove;
  }