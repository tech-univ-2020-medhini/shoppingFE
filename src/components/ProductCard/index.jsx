import React from 'react';
import PropTypes from 'prop-types';
import './index.css';
import Counter from '../Counter';


function ProductCard(props) {
  const {
    id, name, imageLink, price, cart, onClickCounter,
  } = props;
  return (
    <div className="product-card">
      <img src={imageLink} alt={name} />
      <p>
        {name}
      </p>
      <p>
        MRP
        {' '}
        {price}
        {' '}
        /
      </p>
      <Counter cart={cart} onClickCounter={onClickCounter} />
    </div>

  );
}
ProductCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  imageLink: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  cart: PropTypes.number.isRequired,
  onClickCounter: PropTypes.func.isRequired,
};

export default ProductCard;
