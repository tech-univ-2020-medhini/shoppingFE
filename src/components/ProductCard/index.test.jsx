import React from 'react';
import { render } from '@testing-library/react';
import ProductCard from './index';


describe('the product cards', () => {
  it('should render correctly when cardStyle is normal', () => {
    const { asFragment } = render(<ProductCard cardStyle="normal" />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correctly when cardStyle is sold-out', () => {
    const { asFragment } = render(<ProductCard cardStyle="sold-out" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
