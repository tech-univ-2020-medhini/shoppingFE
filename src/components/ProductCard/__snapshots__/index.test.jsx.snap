// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the product cards should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card-sold-out"
  >
    <img />
    <p />
    <p>
      MRP  /
    </p>
    <div
      class="counter"
    >
      <button
        class="counter-button"
        type="button"
      >
        -
      </button>
      <button
        class="counter-button"
        type="button"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
